### PR TITLE
v0.1.0

### PR DESCRIPTION
**Pull request by cloneofsimo**
TODO
- [x] Specify which base model you want to deploy with .env files, quickly deploy them with bash file.
- [x] Refactor inference code with LoRA manager (done here https://github.com/cloneofsimo/lora/pull/172) , because SRP i guess
- [ ] Allow inputs from webui format LoRAs. 
- [x] Better readme so people can launch their own

### CODE REVIEW

File: script/download-weights.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+from transformers import CLIPFeatureExtractor
+
+
+import dotenv
+
+dotenv.load_dotenv()
+
+MODEL_ID = os.environ.get("MODEL_ID", None)
+MODEL_CACHE = "diffusers-cache"
+SAFETY_MODEL_ID = os.environ.get("SAFETY_MODEL_ID", None)
```
Comment (daanelson) [inline script/download-weights.py:18]: We can have some sensible defaults here - `SAFETY_MODEL_ID` can default to `CompVis/stable-diffusion-safety-checker`, `IS_FP16` can default to `0`. 

Also, IMO, we should throw some sort of error if people try to build on top of SD1.5 (and other versions we roll out in the future) where we say "you can just use this existing LoRA" and point them to a URL. It'd be redundant to have, say, 10 different GTAV models that all have the same base weights.

File: cog.yaml [hunk 2]
```diff
@@ -16,8 +16,9 @@ build:
     - "scipy==1.9.3"
     - "transformers==4.25.1"
     - "accelerate==0.15.0"
-    - "git+https://github.com/cloneofsimo/lora.git@848db919c7f3a299bf3d0780ac11cf7cfcb4093e"
+    - "git+https://github.com/cloneofsimo/lora.git"
```
Comment (daanelson) [inline cog.yaml:19]: generally a good idea to keep git repos pinned to a specific commit s.t. we can be sure `cog build` works - but we can update this to the latest commit whenever we're in here! On my list of things to do to have some CI to handle this.

File: test.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+from predict import Predictor
+from inspect import signature
+
+p = Predictor()
+p.setup()
+sig = signature(Predictor.predict)
```
Comment (daanelson) [inline test.py:6]: just want to say again that this is a cool trick :)

File: README.md [hunk 4]
```diff
@@ -5,7 +5,51 @@ Run inference on Replicate:
...
-* Easy-to-use model pre-configured for faces, objects, and styles: [![Replicate](https://replicate.com/replicate/lora-training/badge)](https://replicate.com/replicate/lora-training)
-* Advanced model with all the parameters: [![Replicate](https://replicate.com/replicate/lora-advanced-training/badge)](https://replicate.com/replicate/lora-advanced-training)
+
+- Easy-to-use model pre-configured for faces, objects, and styles: [![Replicate](https://replicate.com/replicate/lora-training/badge)](https://replicate.com/replicate/lora-training)
+- Advanced model with all the parameters: [![Replicate](https://replicate.com/replicate/lora-advanced-training/badge)](https://replicate.com/replicate/lora-advanced-training)
 
 If you have questions or ideas, please join the `#lora` channel in the [Replicate Discord](https://discord.gg/replicate).
+
+## Deployments
+
```
Comment (daanelson) [inline README.md:15]: We should make clear that the default code is https://github.com/replicate/lora-inference; maybe have that as an initial section and then this Manual Deployment work below?

File: script/download-weights.py (hunk 1 above)
Comment (cloneofsimo) [reply script/download-weights.py:18]: Agreed! That really begs another question, how does replicate deal with functional redundancy in the first place? what stops the individual to just remove the error message and upload it anyways?

File: cog.yaml (hunk 2 above)
Comment (cloneofsimo) [reply cog.yaml:19]: Oh nice, ok ill rollback this

File: script/download-weights.py (hunk 1 above)
Comment (cloneofsimo) [reply script/download-weights.py:18]: I think i'll keep saftey model, because that one can change in the future

**Comment by cloneofsimo** [comment]
Since this seems to work, ill merge this as it fixes extra bugs as well. @andreasjansson want to have a look?
https://replicate.com/p/7vzfczi3b5bxlomtficluk5nue

**Pull request merged**